config/initializers/lazy_load_routes.rb

```diff
@@ -0,0 +1,42 @@
+# To lazy load routes:
+# 1) Modify the RoutesReloader to skip normal reload! unless opted-in or Rails server
+# 2) Insert a Rack::Middleware to catch the first request with empty routes, opt-in and reload!
+# 3) Modify the RoutesInspector used by rake routes to detect empty routes, opt-in and reload!
+module RoutesReloaderLazyLoadRoutes
+  def reload!
+    if $force_routes_load || defined?(Rails::Server)
```

Comment:
There any reason we can't change this from a `$global` to a module/class variable on `LazyLoadedRoutes` ?

```ruby
module LazyLoadedRoutes
  class << self
    attr_accessor :force_routes_load
  end
end
```

?

Reply:
yeah, for sure... if this worked for the various tests, I was going to clean it up.

Reply:
but of course, it fails because of rails magic for the URL helper methods, see ☝️

Reply:
yeah, original idea was not to use a global, but we're having issues even getting it to work at all